ganga/GangaCore/Core/GangaRepository/JStreamer.py

```diff
@@ -0,0 +1,196 @@
+"""
+WIP: Replacing the XML files with json equivalent
+
+- JSON Job loader
++ JSON Job dumper
+"""
+
+
+import os
+import json
+
+from GangaCore.Core.exceptions import GangaException
+from GangaCore.Utility.logging import getLogger
+from GangaCore.GPIDev.Base.Proxy import addProxy, stripProxy, isType, getName
+
+from GangaCore.GPIDev.Lib.GangaList.GangaList import GangaList, makeGangaListByRef
+
+# config_scope is namespace used for evaluating simple objects (e.g. File, datetime, SharedDir)
+from GangaCore.Utility.Config import config_scope
+
+from GangaCore.Utility.Plugin import PluginManagerError, allPlugins
+
+from GangaCore.GPIDev.Base.Objects import GangaObject, ObjectMetaclass
+from GangaCore.GPIDev.Schema import Schema, Version
+from GangaCore.GPIDev.Lib.GangaList.GangaList import makeGangaList
+
+# from GangaCore.Core.GangaRepository import SchemaVersionError
+
+import copy
+
+logger = getLogger()
+
+# TODO: Add error handling and logging for the recursion operations 
+class JsonDumper:
+    """Will dump the Job in a JSON file
+    """
+    def __init__(self, dump_location=None):
+        self.errors = []
+        self.dump_location = dump_location
+
+    def parse(self, item):
+        """WIll parse and return the job json
+
+        The received item is a job object with proxy
+        """
+        starting_name, starting_node = "job", item
+        self.job_json = JsonDumper.object_to_json(starting_name, starting_node)
+        
+
+    @staticmethod
+    def object_to_json(name, node):
+        """Will give the attribute information of the provided `node` object as a python dict
+        """
+        node_info = {
+            "name": node._schema.name,
+            "version": node._schema.version.minor,
```

Comment:
Let us discuss if we should take the opportunity to get rid of the schema version attribute. We have in reality never used it.

Reply:
Currently, I do not see any part of the new JStreamer requiring it. If it serves no purpose then I believe no harm lies in removing it.

Reply:
Currently, no part of `JStreamer` requires the use of `version` attr. If it serves no purpose then I believe no harm lies in removing it. What was the reason behind introducing it?

Reply:
It was introduced such that it would be possible to do more advanced upgrading of old jobs when we introduced a new feature. So you could take two old attributes and then maybe merge the information from them into a single new attribute. In reality we have never used such an ability, and in most cases it would just cause confusion.